core/deployment/src/main/java/io/quarkus/deployment/util/AsmUtil.java

```diff
@@ -20,6 +33,36 @@
  */
 public class AsmUtil {
 
+    public static final List<org.objectweb.asm.Type> PRIMITIVES = asList(
+            VOID_TYPE,
+            BOOLEAN_TYPE,
+            CHAR_TYPE,
+            BYTE_TYPE,
+            SHORT_TYPE,
+            INT_TYPE,
+            FLOAT_TYPE,
+            LONG_TYPE);
+    public static final List<org.objectweb.asm.Type> WRAPPERS = asList(
+            getType(Void.class),
+            getType(Boolean.class),
+            getType(Character.class),
+            getType(Byte.class),
+            getType(Short.class),
+            getType(Integer.class),
+            getType(Float.class),
+            getType(Long.class));
+    public static final Map<org.objectweb.asm.Type, org.objectweb.asm.Type> WRAPPER_TO_PRIMITIVE = new HashMap<>();
+
+    static {
+        for (int i = 0; i < AsmUtil.PRIMITIVES.size(); i++) {
+            AsmUtil.WRAPPER_TO_PRIMITIVE.put(AsmUtil.WRAPPERS.get(i), AsmUtil.PRIMITIVES.get(i));
+        }
+    }
+
+    public static org.objectweb.asm.Type autobox(org.objectweb.asm.Type primitive) {
```

Comment:
It would actually convert `void` to `Void` but in this case that can't happen as it's only applied to method parameters in this case.  It's called [here](https://github.com/evanchooly/quarkus/blob/8a535bb37d125799d8beee7c960f6caa269773cb/extensions/panache/hibernate-orm-panache-kotlin/deployment/src/main/java/io/quarkus/hibernate/orm/panache/kotlin/deployment/KotlinPanacheRepositoryClassVisitor.java#L108).

Reply:
and [here](https://github.com/evanchooly/quarkus/blob/8a535bb37d125799d8beee7c960f6caa269773cb/extensions/panache/mongodb-panache-kotlin/deployment/src/main/java/io/quarkus/mongodb/panache/kotlin/deployment/KotlinGenerator.java#L191).